File: RQ1_Test_Generation/OpenAI_Data/HumanEvalPython_output/original/id_104T2000Test.py
# id_104_Test.java
from original.id_104 import unique_digits

import pytest

class id_104_Test:
    """
    Test class of id_104.py.
    It contains ten unit test cases for the :func:`original.id_104.unique_digits`.
    """


    def test_unique_digits_1(self):
        """
        Test case for the function :func:`original.id_104.unique_digits`.
        """
        assert unique_digits([15, 33, 1422, 1]) == [1, 15, 33]

    def test_unique_digits_2(self):
        """
        Test case for the function :func:`original.id_104.unique_digits`.
        """
        assert unique_digits([152, 323, 1422, 10]) == []

    def test_unique_digits_3(self):
        """
        Test case for the function :func:`original.id_104.unique_digits`.
        """
        assert unique_digits([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]) == [1, 3, 5, 7, 9]

    def test_unique_digits_4(self):
        """
        Test case for the function :func:`original.id_104.unique_digits`.
        """
        assert unique_digits([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]) == [1, 3, 5, 7, 9, 11, 13, 15, 17, 19]

    def test_unique_digits_5(self):
        """
        Test case for the function :func:`original.id_104.unique_digits`.
        """
        assert unique_digits([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30]) == [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29]

    def test_unique_digits_6(self):
        """
        Test case for the function :func:`original.id_104.unique_digits`.
        """
        assert unique_digits([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40]) == [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 39]

    def test_unique_digits_7(self):
        """
        Test case for the function :func:`original.id_104.unique_digits`.
        """
        assert unique_digits([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50]) == [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 39, 41, 43, 45, 47, 49]

    def test_unique_digits_8(self):
        """
        Test case for the function :func:`original.id_104.unique_digits`.
        """
        assert unique_digits([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60]) == [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 39, 41, 43, 45, 47, 49, 51, 53, 55, 57, 59]

    def test_unique_digits_9(self):
        """
        Test case for the function :func:`original.id_104.unique_digits`.
        """
        assert unique_digits([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70]) == [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 39, 41, 43, 45, 47, 49, 51, 53, 55, 57, 59, 61, 63, 65, 67, 69]

    def test_unique_digits_10(self):
        """
        Test case for the function :func:`original.id_104.unique_digits`.
        """
        assert unique_digits([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80]) == [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 39, 41, 43, 45, 47, 49, 51, 53, 55, 57, 59, 61, 63, 65, 67, 69, 71, 73, 75, 77, 79]

# id_105.py

def unique_digits(x):
    """Given a list of positive integers x. return a sorted list of all 
    elements that hasn't any even digit.

    Note: Returned list should be sorted in increasing order.
    
    For example:
    >>> unique_digits([15, 33, 1422, 1])
    [1, 15, 33]
    >>> unique_digits([152, 323, 1422, 10])
    []
    """

    odd_digit_elements = []
    for i in x:
        if all (int(c) % 2 == 1 for c in str(i)):
            odd_digit_elements.append(i)
    return sorted(odd_digit_elements)

# id_105_Test.java
from original.id_105 import unique_digits

import pytest

class id_105_Test:
    """
    Test class of id_105.py.
    It contains ten unit test cases for the :func:`original.id_105.unique_digits`.
    """
		
    def test_unique_digits_1(self):
        """
        Test case for the function :func:`original.id_105.unique_digits`.
        """
        assert unique_digits([15, 33, 1422, 1]) == [1, 15, 33]

    def test_unique_digits_2(self):
        """
        Test case for the function :func:`original.id_105.unique_digits`.
        """
        assert unique_digits

File: RQ2_Prompt_Elements/GPT3.5_Data/HumanEvalJava_output/scenario3/id_104T2000Test.java
// UniqueDigits.java
package scenario3;

import java.util.ArrayList;
import java.util.List;

class UniqueDigits {

    /**
     * Given a list of positive integers x. return a sorted list of all
     * elements that hasn't any even digit.
     *
     * Note: Returned list should be sorted in increasing order.
     *
     * For example:
     * >>> uniqueDigits([15, 33, 1422, 1])
     * [1, 15, 33]
     * >>> uniqueDigits([152, 323, 1422, 10])
     * []
     *
     * > uniqueDigits([15, 33, 1422, 1])
     * [1, 15, 33]
     * > uniqueDigits([152, 323, 1422, 10])
     * []
     * > uniqueDigits([12345, 2033, 111, 151])
     * [111, 151]
     * > uniqueDigits([135, 103, 31])
     * [31, 135]
     */
    public static List<Integer> uniqueDigits(List<Integer> x) {
        List<Integer> result = new ArrayList<>();
        for (int i : x) {
            if (hasNoEvenDigit(i)) {
                result.add(i);
            }
        }
        result.sort(null);
        return result;
    }

    private static boolean hasNoEvenDigit(int n) {
        while (n > 0) {
            int digit = n % 10;
            if (digit % 2 == 0) {
                return false;
            }
            n /= 10;
        }
        return true;
    }
}

// UniqueDigitsTest.java
package scenario3;

import java.util.Arrays;
import java.util.List;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
* Test class of {@link UniqueDigits}.
* It contains ten unit test cases for the {@link UniqueDigits#uniqueDigits(List)} method.
*/
class UniqueDigitsTest {

    @Test
    void testEmptyList() {
        List<Integer> input = Arrays.asList();
        List<Integer> expected = Arrays.asList();
        assertEquals(expected, UniqueDigits.uniqueDigits(input));
    }

    @Test
    void testNoEvenDigits() {
        List<Integer> input = Arrays.asList(15, 33, 1422, 1);
        List<Integer> expected = Arrays.asList(1, 15, 33);
        assertEquals(expected, UniqueDigits.uniqueDigits(input));
    }

    @Test
    void testAllEvenDigits() {
        List<Integer> input = Arrays.asList(152, 323, 1422, 10);
        List<Integer> expected = Arrays.asList();
        assertEquals(expected, UniqueDigits.uniqueDigits(input));
    }

    @Test
    void testMixedDigits() {
        List<Integer> input = Arrays.asList(12345, 2033, 111, 151);
        List<Integer> expected = Arrays.asList(111, 151);
        assertEquals(expected, UniqueDigits.uniqueDigits(input));
    }

    @Test
    void testSingleDigit() {
        List<Integer> input = Arrays.asList(5);
        List<Integer> expected = Arrays.asList(5);
        assertEquals(expected, UniqueDigits.uniqueDigits(input));
    }

    @Test
    void testTwoDigits() {
        List<Integer> input = Arrays.asList(13, 24);
        List<Integer> expected = Arrays.asList(13);
        assertEquals(expected, UniqueDigits.uniqueDigits(input));
    }

    @Test
    void testThreeDigits() {
        List<Integer> input = Arrays.asList(135, 103, 31);
        List<Integer> expected = Arrays.asList(31, 135);
        assertEquals(expected, UniqueDigits.uniqueDigits(input));
    }

    @Test
    void testFourDigits() {
        List<Integer> input = Arrays.asList(1234, 5678, 9012);
        List<Integer> expected = Arrays.asList();
        assertEquals(expected, UniqueDigits.uniqueDigits(input));
    }

    @Test
    void testFiveDigits() {
        List<Integer> input = Arrays.asList(12345, 67890, 13579);
        List<Integer> expected = Arrays.asList(13579);
        assertEquals(expected, UniqueDigits.uniqueDigits(input));
    }

    @Test
    void testLargeNumbers() {
        List<Integer> input = Arrays.asList(123456789, 987654321, 246813579);
        List<Integer> expected = Arrays.asList(123456789, 246813579);
        assertEquals(expected, UniqueDigits.uniqueDigits(input));
    }
}